lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+        local brakeClamped = math.Clamp(brake, 0, len)*0.9
+        ph:AddVelocity(velnorm*-brakeClamped)
+
+        if not spin then
+            local cen = ph:GetMassCenter()
+            local fwd = self:GetForward()
+            local lev = ph:GetInertia():Length()
+            local ri = self:GetRight()
+            local ang = self:GetAngles()
+
+            local targetang = yawoffset
+            if trackrotation then
+                targetang = targetang + ent:GetAngles().y
+            end
+            local angdiff = math.AngleDifference(targetang,ang.y)
+            ph:AddAngleVelocity(Vector(0,0,angdiff*phm))
+            ph:AddAngleVelocity(Vector(0,0,ph:GetAngleVelocity().z*-0.3*phm))
+
+            if TRACKING_DEBUG then
+                self.trackingdebugprop:SetAngles(Angle(0,targetang,0))
+            end
+        end
+    end)
+
+    ENT:AddHook("OnRemove", "tracking", function(self)
+        if IsValid(self.trackingdebugprop) then
+            self.trackingdebugprop:Remove()
+        end
+    end)
+
+    ENT:AddHook("FlightControl", "tracking", function(self, ply)
+        if self:GetTracking() then
+            return false
+        end
+    end)
+
+    ENT:AddHook("StopDemat", "tracking", function(self)
+        if self:GetTracking() then
+            self:SetTracking(nil)
+        end
+    end)
+
+    ENT:AddHook("PilotChanged", "tracking", function(self, old, new)
+        if IsValid(new) and self:GetTracking() then
+            self:SendMessage("tracking-pilotwarning", nil, new)
+        end
+    end)
+
+    ENT:AddHook("ShouldTeleportPortal", "tracking", function(self,portal,ent)
+        local trackingEnt = self:GetTracking()
+        if not IsValid(trackingEnt) then return end
+
+        if ent == trackingEnt then
+            return false
+        end
+        if table.HasValue(constraint.GetAllConstrainedEntities(trackingEnt), ent) then
+            return false
+        end
+    end)
+
+    ENT:OnMessage("tracking-set", function(self,data,ply)
+        local ent = data[1]
+
+        if self:GetData("pilot") == ply then
+            self:SetTracking(ent, ply)
+        end
+    end)
+else
+    hook.Add("PostDrawTranslucentRenderables", "tardis-tracking", function()
+        local ext = TARDIS:GetExteriorEnt()
+        if not IsValid(ext) then return end
+    
+        if not ext:GetData("tracking-trace") then return end
+    
+        local pos,ang,ent = ext:GetThirdPersonTrace(LocalPlayer(),LocalPlayer():EyeAngles())
+    
+        local fw = ang:Forward()
+        local bk = fw*-1
+        local ri = ang:Right()
+        local le = ri*-1
+    
+        ext:SetData("tracking-ent",ent)
+        if not IsValid(ent) then
+            local size = 10
+            local col = Color(255,0,0)
+            render.DrawLine(pos,pos+(fw*size),col)
+            render.DrawLine(pos,pos+(bk*size),col)
+            render.DrawLine(pos,pos+(ri*size),col)
+            render.DrawLine(pos,pos+(le*size),col)
+        end
+    end)
```

Comment:
This almost completely duplicates the code from cl_tp_rendering.lua:11-38

It is better to just implement it there with an extra condition 
(just like I did with the destination prop)

Reply:
I thought so too initially and had it implemented as such but it got more complicated because I needed to only draw the crosshair when we weren't looking at an entity, and it needed to set the `tracking-ent` data as well and it got needlessly complicated for the sake of saving a couple lines after adding the conditions to handle it.

It seemed a lot simpler to just do it all here instead even if a couple lines were copied. If you like we could move the render.DrawLine bits into a TARDIS helper function but not sure it's really needed.

Reply:
I really don't like duplicating code like that, it may with a high chance cause confusion in the future

May I suggest b44a7d5dbc4cd3f47b9944d459f2e647b44606cc? (`tracking_tweak` branch)

Reply:
I've merged that with some additional fixes/tweaks on top

https://github.com/MattJeanes/TARDIS/pull/969/commits/60e978a1e8c598799f99a67fc8c94a9a57d8fd25

Reply:
okay!